src/legacy/server/logging/rotate/log_rotator.ts

```diff
@@ -0,0 +1,364 @@
+
+    // cleanup exit listener
+    this._deleteExitListener();
+
+    // stop log file size monitor
+    this._stopLogFileSizeMonitor();
+
+    this.running = false;
+  };
+
+  async _shouldUsePolling() {
+    try {
+      // Setup a test file in order to try the fs env
+      // and understand if we need to usePolling or not
+      const tempFileDir = tmpdir();
+      const tempFile = join(tempFileDir, 'kbn_log_rotation_use_polling_test_file.log');
+
+      await mkdirAsync(tempFileDir, { recursive: true });
+      await writeFileAsync(tempFile, '');
+
+      // setup fs.watch for the temp test file
+      const testWatcher = fs.watch(tempFile, { persistent: false });
+
+      // await writeFileAsync(tempFile, 'test');
+
+      const usePollingTest$ = new Observable<boolean>(observer => {
+        // observable complete function
+        const completeFn = (completeStatus: boolean) => {
+          if (this.stalkerUsePollingPolicyTestTimeout) {
+            clearTimeout(this.stalkerUsePollingPolicyTestTimeout);
+          }
+          testWatcher.close();
+
+          observer.next(completeStatus);
+          observer.complete();
+        };
+
+        // setup conditions that would fire the observable
+        this.stalkerUsePollingPolicyTestTimeout = setTimeout(() => completeFn(true), 15000);
+        testWatcher.on('change', () => completeFn(false));
+        testWatcher.on('error', () => completeFn(true));
+
+        // fire test watcher events
+        setTimeout(() => {
+          fs.writeFileSync(tempFile, 'test');
+        }, 0);
+      });
+
+      // fs.writeFileSync(tempFile, 'test');
+      // await writeFileAsync(tempFile, 'test');
+
+      // wait for the first observable result and consider it as the result
+      // for our use polling test
+      const usePollingTestResult = await usePollingTest$.pipe(first()).toPromise();
+
+      // delete the temp file used for the test
+      await unlinkAsync(tempFile);
+
+      return usePollingTestResult;
+    } catch {
+      return true;
+    }
+  }
+
+  _buildWatchCfg(usePolling: boolean = false) {
+    return {
+      ignoreInitial: true,
+      awaitWriteFinish: false,
+      useFsEvents: false,
+      usePolling,
+      interval: this.pollingInterval,
+      binaryInterval: this.pollingInterval,
+      alwaysStat: true,
+      atomic: false,
+    };
+  }
+
+  async _startLogFileSizeMonitor() {
+    this.usePolling = await this._shouldUsePolling();
+
+    if (this.usePolling && this.usePolling !== this.config.get('logging.rotate.usePolling')) {
+      this.log(
+        ['warning', 'logging:rotate'],
+        'The current environment does not support `fs.watch`. Falling back to polling using `fs.watchFile`'
+      );
+    }
+
+    this.log(
+      ['warning', 'logging:rotate'],
+      'The current environment does not support `fs.watch`. Falling back to polling using `fs.watchFile`'
+    );
+
+    this.stalker = chokidar.watch(this.logFilePath, this._buildWatchCfg(this.usePolling));
```

Comment:
could you inline the config to simplify reading? 
```js
this.stalker = chokidar.watch(this.logFilePath, {
  usePolling,
  interval: this.pollingInterval,
  binaryInterval: this.pollingInterval,
  ignoreInitial: true,
  awaitWriteFinish: false,
  useFsEvents: false,
  alwaysStat: true,
  atomic: false,
});
```

Reply:
👍